File: src/main/java/com/natia/secretmod/utils/FileUtils.java
package com.natia.secretmod.utils;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public class FileUtils {
    public static boolean writeToFile(File fileToWrite, String data) {
        try {
            FileWriter myWriter = new FileWriter(fileToWrite);
            myWriter.write(data);
            myWriter.close();
            return true;
        } catch (IOException e) {
            System.out.println("An error occurred.");
            e.printStackTrace();
        }
        return false;
    }

    public static String readFile(File filetoRead) {
        try {
            File myObj = filetoRead;
            Scanner myReader = new Scanner(myObj);
            while (myReader.hasNextLine()) {
                String data = myReader.nextLine();
                return data;
            }
            myReader.close();
        } catch (FileNotFoundException e) {
            System.out.println("An error occurred when reading file.");
            e.printStackTrace();
        }

        return "";
    }
}

File: src/main/kotlin/natia/skytweaks/utils/FileUtils.kt
package natia.skytweaks.utils

import natia.skytweaks.SkyTweaks

import java.io.File
import java.io.FileNotFoundException
import java.io.FileWriter
import java.io.IOException
import java.util.Scanner

object FileUtils {
    fun writeToFile(fileToWrite: File, data: String): Boolean {
        try {
            val myWriter = FileWriter(fileToWrite)
            myWriter.write(data)
            myWriter.close()
            return true
        } catch (e: IOException) {
            SkyTweaks.LOGGER.info("An error occurred.")
            e.printStackTrace()
        }

        return false
    }

    fun readFile(filetoRead: File): String {
        try {
            val myReader = Scanner(filetoRead)
            while (myReader.hasNextLine()) {
                return myReader.nextLine()
            }
            myReader.close()
        } catch (e: FileNotFoundException) {
            SkyTweaks.LOGGER.info("An error occurred when reading file.")
            e.printStackTrace()
        }

        return ""
    }
}
